components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -324,6 +325,44 @@ class SystemEngineView @JvmOverloads constructor(
             session?.internalNotifyObservers { onContentPermissionRequest(SystemPermissionRequest(request)) }
         }
 
+        override fun onJsAlert(view: WebView?, url: String?, message: String?, result: JsResult): Boolean {
+            session?.apply {
```

Comment:
`session?.apply {}` is very misleading here since you are not really applying something to the session. Since `this` in the block is the `Session` it is really hard to know whether `url` for example accesses the parameter or the url inside `Session`.

If this is just for nullability reasons then I'd prefer something short like:

```Kotlin
val session = session ?: return
```

What do you think?

Reply:
Yes, I did it only for null-availability purpose, it will go with your approach thanks!